content/technical/open-source-as-a-service.md

```diff
@@ -0,0 +1,153 @@
+they're good enough to share with others.
+
+In reality the threat here is far more fundamental: Users of services
+cannot readily change something in the service not only because the
+processes is operated by someone else ... but because such users of
+a service have explicitly chosen not to be involved in operating
+the software.
+
+If the primary way to use [Postgres was as a Service](https://aws.amazon.com/rds/),
+then that (fabulous) project would be starved for contributors. This is
+because the number of contributors in a such a project are a function
+of some small percentage of users deciding to try to make change.
+
+As this mechanism for using software, as a service, takes over the world
+the pool of users who can contribute to Open Source shrinks dramatically.
+
+For a long time I saw this as a fundamental threat that would starve
+Open Source. I was unable reconcile Software as a Service with a healthy
+Open Source ecosystem.
+
+But recently, I've become convinced that if we enable an effective
+contribution model on a service, one that doesn't require that the users
+themselves become operators of the service ... then we can reconcile this
+threat and Open Source will thrive on services instead of be starved by them.
+
+
+# You want me to do what!?
+
+To enable contributions by users of a service, users that have explicitly
+chosen not to operate the software themselves, we have to set up a process
+by which they can make a change to a running service and experience that
+change themselves before others do.
+
+For example, a user of such a hypothetical service should be able to:
+
+ 1. Discover which component of a service to contribute to.
+
+ 2. Make a non-sensical change (like add a printf style log statement) or change the spelling of a word.
+
+ 3. Experience that change when they use the service, or when it acts on their data.
+
+After discussing this with others, I firmly believe we have all the techniques we need,
+whether canary deployments, load balancing, continuous delivery,
+infrastructure as code, and so on. It's been hard to find a single
+problem that an existing practice doesn't solve.
+
+It turns out many engineering teams working have come to that same
+conclusion. Rapid deployment of changes to services is a powerful
+capability highly sought after. For example,
+[GitHub deploys changes before they're merged](https://github.blog/2015-06-02-deploying-branches-to-github-com/).
+Whereas
+[Facebook rapidly deploys changes](https://engineering.fb.com/2017/08/31/web/rapid-release-at-massive-scale/)
+to a few canary machines and scales each change up to production.
+Everyone has their own version of continuous delivery. It's hard to
+take an engineering team seriously if they haven't figured out how
+their team members can get a change rapidly deployed.
+
+As Open Source, we have all the ingredients to have a decisive
+advantage here. To effectively collaborate on services with
```

Comment:
To effectively collaborate on services

-> 

To effectively collaborate on services, **Open Source Services**,